### ISSUE TITLE
Error: minTimeout is greater than maxTimeout

### ISSUE THREAD

**Issue by HelioPC** [raineorshine/npm-check-updates#1155]
- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [x] I am using the latest version of `npm-check-updates`
- [x] I am using `node >= 14`

I'm using the node 18.4.0, ncu 15.0.1, npm 8.12.1, macOs Monterey 12.4 (m1 chip).

---

## Steps to Reproduce

```js
ncu
```

## Current Behavior
<img width="1440" alt="Captura de ecrã 2022-07-02, às 23 29 52" src="https://user-images.githubusercontent.com/59577079/177018130-14e935e0-df11-4fbb-8ec7-e43ee6bcfb06.png">

After updating to version 15.0.1, this error appeared.

**Comment by raineorshine** [comment]
Hi, thanks for reporting.

That's very strange. I am also on OSX 12.4 (m1 chip) and I didn't get this error with the exact same node, npm, and ncu versions.

I looked at the code and the only thing I noticed related to timeouts is that `timeout: undefined` is passed to [pacote](https://github.com/npm/pacote). As far as I can tell this is the same as not passing it at all, but maybe there is some version in the dependency tree that doesn't ignore it and perhaps does a `0 < undefined` comparison with `minTimeout`.

I removed `timeout` when it is undefined, and published to ncu `v15.0.2` if you'd like to give that a try.

Otherwise, it is likely a dependency bug.

**Comment by HelioPC** [comment]
Hey [Raine](https://github.com/raineorshine), sorry for any inconvenience, I just analyzed the source code of the ncu (the files that were throwing the error) and found that the error was in my settings, specifically in `fetch-retry-maxtimeout` and `fetch-retry-mintimeout` whose values were `120000` and `30000` respectively. Although I'm correct (I believe), I reduced their values to `30000` and `3000`, respectively. This didn't occur to me before because before the `ncu` update `15.0.1` everything was working correctly.

**Comment by chungmarcoo** [comment]
same issue here

`node -v `
`v16.17.0`

`ncu --version` 
`16.3.2`

**Comment by raineorshine** [comment]
@chungmarcoo Please post your npm config (tokens elided) and the result of `ncu --verbose` and I can investigate.

**Comment by chungmarcoo** [comment]
``` ncu --verbose ```

```
Initializing
Running in local mode
Finding package file data
Checking /Users/marco/package.json

Options:
{
  args: [],
  cache: false,
  cacheExpiration: 10,
  cacheFile: '~/.ncu-cache.json',
  cacher: undefined,
  cli: true,
  concurrency: 8,
  deep: false,
  dep: 'prod,dev,bundle,optional',
  errorLevel: 1,
  format: [],
  json: false,
  loglevel: 'verbose',
  minimal: false,
  packageManager: 'npm',
  prefix: undefined,
  reject: [],
  retry: 3,
  target: 'latest',
  verbose: true,
  workspace: []
}

Current versions:
{ 'super-tinypng': '^1.0.5' }

Fetching latest versions
/Users/marco/.nvm/versions/node/v16.17.0/lib/node_modules/npm-check-updates/build/src/index.js:56
    throw err;
    ^

Error: minTimeout is greater than maxTimeout
    at Object.exports.timeouts (/Users/marco/.nvm/versions/node/v16.17.0/lib/node_modules/npm-check-updates/node_modules/retry/lib/retry.js:29:11)
    at Object.exports.operation (/Users/marco/.nvm/versions/node/v16.17.0/lib/node_modules/npm-check-updates/node_modules/retry/lib/retry.js:4:26)
    at promiseRetry (/Users/marco/.nvm/versions/node/v16.17.0/lib/node_modules/npm-check-updates/node_modules/promise-retry/index.js:23:23)
    at remoteFetch (/Users/marco/.nvm/versions/node/v16.17.0/lib/node_modules/npm-check-updates/node_modules/make-fetch-happen/lib/remote.js:48:10)
    at fetch (/Users/marco/.nvm/versions/node/v16.17.0/lib/node_modules/npm-check-updates/node_modules/make-fetch-happen/lib/fetch.js:99:13)
    at makeFetchHappen (/Users/marco/.nvm/versions/node/v16.17.0/lib/node_modules/npm-check-updates/node_modules/make-fetch-happen/lib/index.js:10:10)
    at doFetch (/Users/marco/.nvm/versions/node/v16.17.0/lib/node_modules/npm-check-updates/node_modules/npm-registry-fetch/lib/index.js:108:15)
    at async RegistryFetcher.packument (/Users/marco/.nvm/versions/node/v16.17.0/lib/node_modules/npm-check-updates/node_modules/pacote/lib/registry.js:92:19)
    at async viewMany (/Users/marco/.nvm/versions/node/v16.17.0/lib/node_modules/npm-check-updates/build/src/package-managers/npm.js:233:18)
    at async viewMany (/Users/marco/.nvm/versions/node/v16.17.0/lib/node_modules/npm-check-updates/build/src/package-managers/npm.js:237:31)
```

**Comment by raineorshine** [comment]
@chungmarcoo And your npm config please?

**Comment by chungmarcoo** [comment]
> @chungmarcoo And your npm config please?

```
; "user" config from /Users/marco/.npmrc

fetch-retry-maxtimeout = 120000 
fetch-retry-mintimeout = 20000 

; node bin location = /Users/marco/.nvm/versions/node/v16.17.0/bin/node
; node version = v16.17.0
; npm local prefix = /Users/marco
; npm version = 8.19.2
; cwd = /Users/marco
; HOME = /Users/marco
; Run `npm config ls -l` to show all defaults.
```

**Comment by raineorshine** [comment]
Thank you. I was able to reproduce the issue, and found that [libnpmconfig](https://github.com/npm/libnpmconfig) does not parse the types of config values. So `fetch-retry-maxtimeout` was being set to `'120000'` and `fetch-retry-mintimeout` was being set to `'20000'`, and of course `'120000' > '2000'` is false.

Fixed and published in `v16.3.5`.

**Comment by chungmarcoo** [comment]
> 

Happy to hear your update. I will try the update tomorrow, appreciate for your work and have a good day.
